src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java

```diff
@@ -4,24 +4,25 @@
 import mate.academy.spring.dao.UserDao;
 import mate.academy.spring.model.User;
 import mate.academy.spring.service.UserService;
-import mate.academy.spring.util.HashUtil;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
     private static final int SALT_LENGTH = 10;
+    private final PasswordEncoder passwordEncoder;
     private final UserDao userDao;
 
-    public UserServiceImpl(UserDao userDao) {
+    public UserServiceImpl(PasswordEncoder passwordEncoder,
+                           UserDao userDao) {
+        this.passwordEncoder = passwordEncoder;
         this.userDao = userDao;
     }
 
     @Override
     public User add(User user) {
-        String salt = HashUtil.getSalt(SALT_LENGTH);
-        String securePassword = HashUtil.generateSecurePassword(user.getPassword(), salt);
-        user.setPassword(securePassword);
-        user.setSalt(salt);
+        String encodedPassword = passwordEncoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
```

Comment:
```suggestion
        user.setPassword(passwordEncoder.encode(user.getPassword()));
```
Don't create redundant variables